### PR TITLE
optimize: ehance metrics and fix seata-server UT not work 

### LINKED ISSUE #3903: seata-server UT not work
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description

image

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- OS :
- Others:

### PR DESCRIPTION
**Pull request by slievrly**
### Ⅰ. Describe what this PR did

optimize: fix seata-server UT not work 

fixes #3903   #3993 

https://maven.apache.org/surefire/maven-surefire-plugin/examples/junit-platform.html

### CODE REVIEW

File: core/src/main/java/io/seata/core/event/GuavaEventBus.java [hunk 1]
```diff
@@ -32,6 +34,7 @@
 public class GuavaEventBus implements EventBus {
     private static final Logger LOGGER = LoggerFactory.getLogger(GuavaEventBus.class);
     private final com.google.common.eventbus.EventBus eventBus;
+    private static Set<Object> subscriberSet = new HashSet<>();
```
Comment (a364176773) [inline core/src/main/java/io/seata/core/event/GuavaEventBus.java:37]: ConcurrentHashMap.newKeySet();

Comment (slievrly) [reply core/src/main/java/io/seata/core/event/GuavaEventBus.java]: fixed

**Review by a364176773 — APPROVED** [review]
LGTM

File: pom.xml [hunk 2]
```diff
@@ -140,6 +140,12 @@
             <version>${junit-jupiter.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (l81893521) [inline pom.xml:143]: Look like 
```
jnit-jupiter-engine
```
already include
```
jnit-jupiter-api
```
and the dependency of junit-jupiter-api and junit-jupiter-engine can change to 
```
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter</artifactId>
    <version>${junit-jupiter.version}</version>
    <scope>test</scope>
</dependency>
```
The dependency diagram
image

Comment (slievrly) [reply pom.xml]: fixed

File: server/src/test/java/io/seata/server/lock/db/DataBaseLockManagerImplTest.java [hunk 3]
```diff
@@ -38,6 +42,8 @@
 /**
  * @author zhangsen
  */
+@SpringBootTest
```
Comment (l81893521) [inline server/src/test/java/io/seata/server/lock/db/DataBaseLockManagerImplTest.java:45]: When I check the spring boot document I found that there is no need to add @ExtendWith(SpringExtension.class) when use junit 5 and @SpringBootTest already has it.
image

Comment (slievrly) [reply server/src/test/java/io/seata/server/lock/db/DataBaseLockManagerImplTest.java:43]: fixed

File: server/src/test/java/io/seata/server/store/db/LogStoreDataBaseDAOTest.java [hunk 4]
```diff
@@ -27,6 +27,10 @@
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.extension.ExtendWith;
```
Comment (l81893521) [inline server/src/test/java/io/seata/server/store/db/LogStoreDataBaseDAOTest.java:30]: Please remove the unused import

Comment (slievrly) [reply server/src/test/java/io/seata/server/store/db/LogStoreDataBaseDAOTest.java]: fixed.

**Review by l81893521 — APPROVED** [review]
LGTM

**Pull request merged**